sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldUseBaseTypes.cs

```diff
@@ -286,7 +289,8 @@ public Diagnostic GetRuleViolation()
                 var mostGeneralType = FindMostGeneralType();
 
                 if (!Equals(mostGeneralType, parameterSymbol.Type) &&
-                    CanSuggestBaseType(mostGeneralType.GetSymbolType()))
+                    IsConsistentAccessibility(mostGeneralType.GetEffectiveAccessibility(), methodAccessibility) &&
```

Comment:
My comment seem to disappear...
The logic here is subtly wrong. It finds the most general type, then checks if it is accessible. Correct way would be to make accessibility checks as we go.

Don't think it's worth addressing, but a test or at least a comment would be useful.